### ISSUE TITLE
Add flags to UsesPermissions annotation

### ISSUE THREAD

**Issue by puravidaapps** [mit-cml/appinventor-sources#2738]
Please update the UsesPermissions annotation and add the following flags

1) android:maxSdkVersion
Example:
```
<uses-permission android:name="android.permission.BLUETOOTH_ADMIN"
                     android:maxSdkVersion="30" />

```
2) android:usesPermissionFlags
Example:
```
<uses-permission android:name="android.permission.BLUETOOTH_SCAN"
                     android:usesPermissionFlags="neverForLocation" />
```

see also https://developer.android.com/guide/topics/connectivity/bluetooth/permissions

Taifun

**Comment by ewpatton** [comment]
Actually, this is going to be needed to fix issues with both the builtin Bluetooth components as well as the BLE extension, so I'm going to increase the priority on this.

**Comment by ewpatton** [comment]
@puravidaapps If you try to build an extension using #2754, you should be able to test it on this server: http://nb190a-bugfixes.ai2-ewpatton-temp.appspot.com/. Let me know if you run into any problems.

**Comment by shreyashsaitwal** [comment]
@ewpatton What do you think about allowing extensions to have an `AndroidManifest.xml` packaged with them (like AARs) and getting it merged with the main manifest during compilation using the [manifest merger tool](https://mvnrepository.com/artifact/com.android.tools.build/manifest-merger)?

This will free extension devs from depending on AI2 to support new manifest elements.

**Comment by ewpatton** [comment]
It is something to consider. While the annotations have mostly tracked the AndroidManifest structure, one potential benefit of having a structure under our own control is that we can also extend it to support extensions for iOS when the time comes (we won't be able to hot-load code, but compiled apps could benefit).

**Comment by shreyashsaitwal** [comment]
> one potential benefit of having a structure under our own control is that we can also extend it to support extensions for iOS

I've never done iOS development, but assuming the iOS equivalent of the Android manifest isn't exactly a 1-on-1 mapping, won't this be more of a hassle since you'd then have to make sure that every Android manifest annotation translates well to the iOS equivalent?

**Comment by ewpatton** [comment]
One other thing that the annotations provide that the AndroidManifest approach won't is that we can conditionally include permissions etc based on which blocks are included. Therefore, we would have to continue to support both approaches.

**Comment by shreyashsaitwal** [comment]
The Android Gradle plugin allows using various XML attributes in the `tools` namespace to reconfigure the behavior of the manifest merger tool. Using the same technique, we can introduce a new attribute, say `tools:includeIfUsed`, which we could then use to process conditionals.

```xml
<permission tools:includeIfUsed="<method-name>" android:name="<permission-name>" />
```

**Comment by ewpatton** [comment]
Resolved by 4d6f522fbbf41c01691fe3d513726b490c215c33
